app/controllers/pieces_controller.rb

```diff
@@ -6,7 +6,14 @@ def index
 
   def update
     piece = Piece.find(params[:id])
-    piece.update_attributes(piece_params)
+    Rails.logger.debug "PIECE Controller Update. call move_to in pieces model"
+
+    if piece.move_to!(piece_params[:row].to_i, piece_params[:col].to_i)
+      Rails.logger.debug "VALID MOVE. Update Record - SQL update in server window"
+      piece.update_attributes(piece_params)
```

Comment:
This should move back into `move_to!` so that the update actually happens in the `move_to!` method (otherwise `move_to!` is essentially the same as `valid_move?`).

You can still pass the return value though (see below):